app/scripts/lib/experiment.js

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'lib/experiments/open-gmail',
+  'lib/experiments/mailcheck',
+  'lib/experiments/coppa-view',
+  'lib/url'
+], function (_, OpenGmailExperiment, MailcheckExperiment, CoppaExperiment, Url) {
+  'use strict';
+
+  var CHOOSE_ABLE_EXPERIMENT = 'chooseAbExperiment';
+  var FORCE_EXPERIMENT_PARAM = 'forceExperiment';
+  // all available experiments that must be independent for A/B testing purposes.
+  var EXPERIMENTS = {
```

Comment:
> Do you ever foresee other types of experiments being mutually exclusive

All experiments should be mutually exclusive, unless it is something like "show avatar change button" which is not really an experiment but still uses able.